### ISSUE TITLE
HTTP eventsource  GET / hr_streaming POST mass call

### ISSUE THREAD

**Issue by cedbale** [primus/primus#635]
**Version:**
6.1.0

**Environment:**
  - **Operating system:**
server Debian 8

  - **Browser:**
Mac OS X 10.10 / Firefox 58.0

  - **Node.js:** 
v6.11.3

**Steps to reproduce:**
not able to reproduce - this behavior appear sometimes

Hello guys,

I recently saw in my Nginx (with a `nodejs`/`express`/`primus`/ `sockJS` as upstream over HTTPS) logs that sometimes Primus browser clients spam my server logs with a lot of http call. The log extract below is made multiples times per second. 
After having a look at Primus source code, [i saw a reference](https://github.com/primus/primus/blob/7f908722f16e799a96304e8218d94a7cfd7c408d/transformers/sockjs/library.js#L2771) on `/xhr_streaming` to CORS cross browser compatibility check for XHR but i am not sure that this behavior refer to this as i am not concern with CORS case. All HTTP calls are made from the same origin.

```
x.x.x.x - - [27/Feb/2018:21:11:07 +0000] "GET /primus/589/vlnmgpne/eventsource?_user=3067bf39-f689-4142-a84d-ab550586d7fc&name=username&client=web&_primuscb=M7Nv3j1 HTTP/1.1" 200 52 "https://iamthereferer.com" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:58.0) Gecko/20100101 Firefox/58.0"
x.x.x.x - - [27/Feb/2018:21:11:07 +0000] "POST /primus/589/kh1wldtk/xhr_streaming?_user=3067bf39-f689-4142-a84d-ab550586d7fc&name=username&client=web&_primuscb=M7Nv3j1&t=1519765801749 HTTP/1.1" 200 2092 "https://iamthereferer.com" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:58.0) Gecko/20100101 Firefox/58.0"
x.x.x.x - - [27/Feb/2018:21:11:07 +0000] "POST /primus/589/kh1wldtk/xhr_streaming?_user=3067bf39-f689-4142-a84d-ab550586d7fc&name=username&client=web&_primuscb=M7Nv3j1&t=1519765801816 HTTP/1.1" 200 2092 "https://iamthereferer.com" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:58.0) Gecko/20100101 Firefox/58.0"
x.x.x.x - - [27/Feb/2018:21:11:07 +0000] "POST /primus/589/kh1wldtk/xhr_streaming?_user=3067bf39-f689-4142-a84d-ab550586d7fc&name=username&client=web&_primuscb=M7Nv3j1&t=1519765801885 HTTP/1.1" 200 2092 "https://iamthereferer.com" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:58.0) Gecko/20100101 Firefox/58.0"
x.x.x.x - - [27/Feb/2018:21:11:07 +0000] "POST /primus/589/kh1wldtk/xhr_streaming?_user=3067bf39-f689-4142-a84d-ab550586d7fc&name=username&client=web&_primuscb=M7Nv3j1&t=1519765801952 HTTP/1.1" 200 2092 "https://iamthereferer.com" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:58.0) Gecko/20100101 Firefox/58.0"

```

Can anyone give me a tip to understand if this is a normal behavior and why this happen ? 

In a normal usage of Websocket with Primus i did never see those http call before.

thank you for the help

**Comment by lpinca** [comment]
The code you referenced it's actually sockjs code, we embed that in our client.
It may be a sockjs issue but it's hard to tell. Try to disable some fallback transports in sockjs and see if the issue is limited to xhr_streaming and search their issue tracker for something similar.

**Comment by bard** [comment]
I'm also seeing xhr_streaming spam (to DOS levels). Related: https://github.com/sockjs/sockjs-client/issues/308

Would it be very hard to rebuild Primus with the current (1.1.4+) SockJS client?

**Comment by lpinca** [comment]
No it's very easy, there is a script for it, but we didn't update because not much changed since version 1.1.2, see https://github.com/sockjs/sockjs-client/compare/v1.1.2...v1.1.4

It might make sense to disable the affected transport.

**Comment by bard** [comment]
Ah, the fix is only in master: https://github.com/sockjs/sockjs-client/commit/eef1957a4d8007a64e170a4bbb2c635e40d870b2

Applying it fixed the problem for me.

**Comment by lpinca** [comment]
@bard if a new `sockjs-client` version is not on the horizon, we can float the patch. We have done it in the past for both engine.io and sockjs. It's just a matter of putting the patch here https://github.com/primus/primus/tree/master/transformers/sockjs/patches and calling the update script.

**Comment by lpinca** [comment]
@VivienAdnot the patch is on the master branch of sockjs-client but it has not yet been included in a release or it would have been easier for us to update.

**Comment by lpinca** [comment]
I've asked @brycekahle if they can cut a new release: https://github.com/sockjs/sockjs-client/commit/eef1957a4d8007a64e170a4bbb2c635e40d870b2#commitcomment-29161994.

**Comment by brycekahle** [comment]
sockjs-client `1.1.5` released

**Comment by lpinca** [comment]
Fixed by 041f3abe7c3c34cbf7b820df693b00c211f48b1c.

**Comment by cedbale** [comment]
thank you @lpinca @brycekahle
